x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/fleet/agent_details_page/components/agent_details.tsx

```diff
@@ -69,8 +74,39 @@ export const AgentDetailsContent: React.FunctionComponent<{
           description:
             typeof agent.local_metadata.elastic === 'object' &&
             typeof agent.local_metadata.elastic.agent === 'object' &&
-            typeof agent.local_metadata.elastic.agent.version === 'string'
-              ? agent.local_metadata.elastic.agent.version
+            typeof agent.local_metadata.elastic.agent.version === 'string' ? (
+              <EuiFlexGroup gutterSize="s" alignItems="center" style={{ minWidth: 0 }}>
+                <EuiFlexItem grow={false} className="eui-textNoWrap">
+                  {agent.local_metadata.elastic.agent.version}
+                </EuiFlexItem>
+                {isAgentUpgradeable(agent, kibanaVersion) ? (
+                  <EuiFlexItem grow={false}>
+                    <EuiText color="subdued" size="s" className="eui-textNoWrap">
+                      <EuiIcon size="m" type="alert" color="warning" />
+                      &nbsp;
+                      <FormattedMessage
+                        id="xpack.ingestManager.agentList.agentUpgradeLabel"
+                        defaultMessage="Upgrade available"
+                      />
+                    </EuiText>
+                  </EuiFlexItem>
+                ) : null}
+              </EuiFlexGroup>
+            ) : (
+              '-'
+            ),
+        },
+        {
+          title: i18n.translate('xpack.ingestManager.agentDetails.releaseLabel', {
+            defaultMessage: 'Agent release',
+          }),
+          description:
+            typeof agent.local_metadata.elastic === 'object' &&
+            typeof agent.local_metadata.elastic.agent === 'object' &&
+            typeof agent.local_metadata.elastic.agent.snapshot === 'boolean'
+              ? agent.local_metadata.elastic.agent.snapshot === true
+                ? 'snapshot'
+                : 'stable'
```

Comment:
in case of not snapshot the key might be missing which results in `-` in UI not sure it's ok

Reply:
Is it possible to send snapshot is false if it is not a snapshot?  Otherwise how do i differentiate between it not being a snapshot because the key doesn't exist and the key not existing because the agent version they are on does not send snapshot information?